models/result.go

```diff
@@ -58,6 +58,26 @@ func GetResultById(id int64) (*Result, error) {
 	return result, nil
 }
 
+// GetResultByIdWithRelations retrieves Result by Id with assigned relations. Returns error if Id doesn't exist
+func GetResultByIdWithRelations(id int64) (*Result, error) {
```

Comment:
I need to add this because the existing `GetResultById` function does not assign the associated `Links` and `AdLinks` 
I also try the `RelatedSel` function from [Beego's ORM](https://beego.me/docs/mvc/model/query.md#relatedsel) but it doesn't work on the reverse relation like a many to one relation 😓